processing/src/main/java/io/druid/query/BaseQuery.java

```diff
@@ -67,6 +70,61 @@
     return parseInt(query, "uncoveredIntervalsLimit", defaultValue);
   }
 
+  public static <T> Map<String, Object> getContextTypeHints(Query<T> query)
```

Comment:
I've removed this function now, based on @gianm's comment about moving the type specification to DimensionSpec